contracts/TakeBack.sol

```diff
@@ -0,0 +1,74 @@
+pragma solidity ^0.4.23;
+
+import 'zeppelin-solidity/contracts/token/ERC20/ERC20.sol';
+import 'zeppelin-solidity/contracts/ownership/Ownable.sol';
+
+contract Claim is Ownable{
+
+    using StringHelper for *;
+
+
+    // address of RING.sol on ethereum
+    address public ring;
+
+    mapping (address => uint) public userToNonce;
+
+
+    // used for old&new users to claim their ring out
+    event TakeBack(address indexed _user, uint indexed _nonce);
+    // used for owner to claim all kind of token
+    event ClaimedTokens(address indexed _token, address indexed _controller, uint _amount);
+
+
+    // _hashmessage = hash("${_user}${_nonce}${_value}")
+    // _v, _r, _s are from owner's signature on _hashmessage
+    // claimRing(...) is invoked by the user who want to claim rings
+    // while the _hashmessage is signed by owner
+    function takeBack(uint _nonce, uint _value, bytes32 _hashmessage, uint8 _v, bytes32 _r, bytes32 _s) public {
+
+        address _user = msg.sender;
+
+        // verify the _nonce is right
+        require(userToNonce[_user] == _nonce);
+
+        // verify the _hashmessage is signed by owner
+        require(owner == verify(_hashmessage, _v, _r, _s));
```

Comment:
For safety concern, we could add another supervisor address different with owner, because supervisor address private key will be kept in server node and less secure.